index.php

```diff
@@ -266,6 +266,12 @@
         $cache->save($cfgCache);
     }
     $cfg = $cfgCache->get();
+
+    // TODO Reload language
```

Comment:
As far as I can see / remember from the other PR the new language is passed as either $_GET or $_POST parameter and thus can be used to trigger cache flush / reload, see https://github.com/lansuite/lansuite/pull/481/commits/f9441a7e48e6e240aef68ad358e085bb5d1b719f#diff-e346d84a075b7d32e92ee9bd65971289b56a8fc7d80c836d68ac8afb19b72374

Could that be what is needed here?

Reply:
The reload is needed because the database is initialized AFTER the language is retrieved from GET/POST. The function you linked also checked for $cfg['sys_language'], which contains the configured language from the user.

A simple `get_lang()` was not sufficient during testing. This will set the language but don't reload the lang_cache.
I will look into this deeper, when we rework the translation class. Thats why I left https://github.com/lansuite/lansuite/issues/415 open

Reply:
I meant this code block here within `get_lang()`
```
        if (isset($_POST['language']) && $_POST['language']) {
            $_SESSION['language'] = $_POST['language'];
            $this->lang_cache = array();
        } elseif (isset($_GET['language']) && $_GET['language']) {
            $_SESSION['language'] = $_GET['language'];
            $this->lang_cache = array();
```
The invalidates `lang_cache` which should then trigger it to reload it from source, which is then the DB

Reply:
Ah, I see.
I am not sure, if I prefer such a "side effect", because this function is called `get...` and not `clear...` or `refresh...`.
It won't be obvious that this function clears/resets the cache.

I would rather think about a method like "loadTranstrations()" or "refreshTranslations()" of something like this.